src/components/App.js

```diff
@@ -1,17 +1,29 @@
-import React from 'react';
+import React, { useState } from 'react';
 import '../styles/App.scss';
 import '../styles/Button.scss';
 import '../styles/ButtonPanel.scss';
 import ButtonPanel from './ButtonPanel';
 import Display from './Display';
+import calculate from '../logic/calculate';
 
 function App() {
+  const [total, setTotal] = useState(null);
+  const [next, setNext] = useState(null);
+  const [operation, setOperation] = useState(null);
+
+  const handleClick = buttonName => {
+    let data = calculate({ total, next, operation }, buttonName);
+
+    setNext(data.next);
+    setOperation(data.operation);
+    setTotal(data.total);
+  };
+
   return (
     <div id='container'>
-      <Display />
-      {/* Pass result="" props to the display component */}
+      <Display result={next ? next : total} />
```

Comment:
Instead of doing a ternary here, we can just set another state called `result` and do the necessary operation on it using `setResult`:

```javascript
const [result, setResult] = useState(null) 
```

Then in the `handleClick` function:

```javascript
setResult(data.next || data.total)
```